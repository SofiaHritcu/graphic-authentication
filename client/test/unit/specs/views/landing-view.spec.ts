import { shallowMount } from "@vue/test-utils";
import LandingView from "@/views/landing-view.vue";
import "regenerator-runtime/runtime";

describe("Landing View", () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallowMount(LandingView, {
      mocks: {
        $router: {
          push: jest.fn(),
        },
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  describe("snapshot", () => {
    it("should match snapshot", () => {
      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });

  describe("methods", () => {
    describe("handleButtonClick", () => {
      it("should execute router push with the right parameters", async () => {
        expect.assertions(1);
        const spyRouterPush = jest.spyOn(wrapper.vm.$router, "push");

        await wrapper.vm.handleButtonClick("ga-login");

        expect(spyRouterPush).toHaveBeenCalledWith({ name: "ga-login" });
      });
    });
  });
});
