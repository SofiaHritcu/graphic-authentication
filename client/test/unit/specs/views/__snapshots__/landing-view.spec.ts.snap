// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing View snapshot should match snapshot 1`] = `
<v-row
  align-content="center"
  class="ga__home_view"
>
  <v-col
    class="ga__home_view__left"
    cols="10"
    md="5"
    sm="5"
  />
   
  <v-col
    class="ga__home_view__right"
    cols="12"
    md="7"
    sm="7"
  >
    <div
      class="text-md-h4 grey--text text--darken-1"
    >
      Here is your
    </div>
     
    <div
      class="text-overline font-weight-light white--text ga__home_view__right__title"
    >
      
      IPass
    
    </div>
     
    <div>
      <v-btn
        class="mt-3 mb-3"
        color="teal lighten-2"
        outlined=""
        x-large=""
      >
        Sign Up
      </v-btn>
    </div>
     
    <div>
      <v-btn
        class="mb-3"
        color="deep-orange lighten-1"
        outlined=""
        x-large=""
      >
        Log In
      </v-btn>
    </div>
     
    <div>
      <v-btn
        color="yellow lighten-2"
        outlined=""
        x-large=""
      >
        Learn More
      </v-btn>
    </div>
  </v-col>
</v-row>
`;
